sotodlib/io/hkdb.py

```diff
@@ -70,19 +70,22 @@ def __post_init__(self):
         if self.db_file is not None:
             self.db_url = f"sqlite:///{self.db_file}"
 
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "HkConfig":
```

Comment:
Please docstring for this... or perhaps better would be explain, in the class docstring, what this lets you get away with.

Reply:
Thanks Matthew, I added some additional info to the docstrings about when you might want to use this.

Reply:
I was hoping for something along the lines of
"db_url can be a string, or else a dict (which will be passed to db.URL.create()).  In either case, environment variables will be expanded."

Reply:
I see. Some of the awkwardness here comes from type-checking, and the fact that the type of the dataclass field "db_url" will always be of type `Union[str, db.URL]`, but it can be initialized a number of different ways.

These days I will generally avoid using dataclasses in these cases, and just use different type signatures for the `__init__` arguments and the class attributes. Would you rather I just change the implementation here so the init function accepts type Dict, and the behavior can be documented in the Args section?

Reply:
I don't find this particularly awkward, but maybe I just don't understand it. The comments you added are enough for me!